jbmc/src/java_bytecode/java_entry_point.cpp

```diff
@@ -307,18 +307,25 @@ bool is_java_main(const symbolt &function)
   return named_main && is_static && has_correct_type && public_access;
 }
 
-///  Extends \p init_code with code that allocates the objects used as test
-///  arguments for the function under test (\p function) and
-///  non-deterministically initializes them.
-///
-///  All the code generated by this function goes to __CPROVER__start, just
-///  before the call to the method under test.
-///
-///  \returns A std::vector of symbol_exprt, one per parameter of \p function,
-///  containing the objects that can be used as arguments for \p function.
-exprt::operandst java_build_arguments(
+/// \param function: The function under test, for which to build the arguments.
+/// \param symbol_table: For the purposes of adding any local variables
+///   generated or any functions which are generated and called.
+/// \param assume_init_pointers_not_null: If this is true then any reference
+///   type parameters to the function under tests are assumed to be non-null.
+/// \param object_factory_parameters: Configuration of the object factory.
+/// \param pointer_type_selector: Means of selecting the type of value
+///   constructed for reference types which are initialised by the code
+///   returned.
+/// \returns A pairing of the code to initialise the arguments and a std::vector
+///   of the expressions for these arguments. The vector contains one element
+///   per parameter of \p function. The vector of expressions can be used as
+///   arguments for \p function. The code returned allocates the objects used as
+///   test arguments for the function under test (\p function) and
+///   non-deterministically initializes them. This code returned must be placed
+///   into the code block of the function call, before the function call.
+///   Typically this code block would be  __CPROVER__start.
+std::pair<code_blockt, std::vector<exprt>> java_build_arguments(
```

Comment:
Consider using a struct rather than a pair so you can call these things by name

Reply:
Then requires less documentation since the types do the talking.